### PR TITLE
CONSOLE-3565: Expose annotations modal in dynamic plugins sdk

### PR DESCRIPTION
**Pull request by TheRealJon**
- Factor out AnnotationsModal component from annotationsModal launcher
- Factor out ModalWrapper from createModalLauncher and export as internal API from SDK
- Move ModalWrapperProps, TagsModalProps, and AnnotationsModalProps to SDK internal types
- Implement useAnnotationsModal hook in dynamic plugin SDK and use in core details page component
- Improve SDK ModalComponent prop types
- Fix useModal hook example in SDK docs
- Address some minor tech debt issues in modal factory

### CODE REVIEW

**Comment by TheRealJon** [comment]
Note I'm using the PatternFly modal to wrap the internal AnnotationModal component which causes slight style issues, but this is one of the less complex and risky solutions. Here are some other approaches I considered:

- Factor out a ModalWrapper component from the createModalLauncher func so that we can use it as a wrapper for the SDK modal. (low risk and complexity, but leaves tech debt)
- Refactor the internal AnnotationsModal to use the PF modal (higher complexity and risk, but good for PF adoption)
- Build a new modal component in the SDK using the PF modal API, and use the internal NameValueEditor component as the body. (slightly more risk than current implementation, adds some repetition since the internal annotation modal would still exist, good for PF adoption)

**Comment by jhadvig** [comment]
> Build a new modal component in the SDK using the PF modal API, and use the internal NameValueEditor component as the body. (slightly more risk than current implementation, adds some repetition since the internal annotation modal would still exist, good for PF adoption)

Not sure why we cant just remove the internal annotation modal once we build the new  one in SDK and use that new one internally as well.

**Comment by jhadvig** [comment]
@TheRealJon I think I like the third suggestion the most, if we could avoid the code duplicity.

**Comment by TheRealJon** [comment]
@jhadvig The SDK modal implementation is different than our internal one, and there isn't a good way to migrate without some refactoring. That's why I say we'll have to leave the legacy one in place until we can fully migrate to the SDK modal system. The other issue is with styles and layout. The PF modal looks different than our existing internal modal. Mainly, the save/cancel buttons are on the left. That being said, we could just update the internal modal to have left-aligned buttons too. This would make the PF transition a little easier later.

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/modals/useAnnotationsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * A hook for launching a modal for editing a resource's annotations.
+ *
+ * @hook useAnnotationsModal
+ * @argument resource - The resource to edit annotations for.
+ * @returns a function which will launch a modal for editing a resource's annotations.
+ * @example
+ * const PodAnnotationsButton = ({ pod }) => {
+ *   const launchAnnotationsModal = useAnnotationsModal<PodKind>(pod);
+ *   return <button onClick={launchAnnotationsModal}>Edit Pod Annotations</button>
```
Comment (rhamilto) [inline frontend/packages/console-dynamic-plugin-sdk/src/app/components/modals/useAnnotationsModal.tsx:31]: Add i18n to the button label to encourage its usage?

**Review by rhamilto — COMMENTED** [review]
Nice cleanup!  One question about the doc example, but otherwise LGTM.

**Comment by TheRealJon** [comment]
/
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

**Comment by yanpzhan** [comment]
@TheRealJon @jhadvig If regression test about add/edit/delete annotation from edit annotation modal of an resource(eg, a deployment) on console is enough for the pr? Current regression test passed. If no additional checkpoint I would accept the pr.

**Pull request merged**